### ISSUE TITLE
KNeighboursRegressor converter issues

### ISSUE THREAD

**Issue by prabhat00155** [onnx/sklearn-onnx#204]
1) class probability mismatches with int features:
X, y = make_regression(n_features=10, n_samples=1000, random_state=42)
X = X.astype(np.int64)

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.5, random_state=42)
model = KNeighborsRegressor().fit(X_train, y_train)

onnx_model = convert_sklearn(model, 'lr', [('input', Int64TensorType([1, X_test.shape[1]]))])
save_model(onnx_model, 'lr2.onnx')
sess = InferenceSession('lr2.onnx')
res = []
for x in X_test:
    res.append(sess.run(None, input_feed={'input': x.astype(np.int64)}))

print(np.mean(np.isclose(model.predict(X_test), list(map(lambda x: x[0][0][0], res)), rtol=1e-5)))

0.264

2) Probability score mismatches with the following combination of (algorithm, metric) hyper-parameters:

ball_tree, p | ball_tree, chebyshev | ball_tree, infinity | ball_tree, hamming | ball_tree, canberra | ball_tree, braycurtis | ball_tree, matching | ball_tree, jaccard | ball_tree, dice | ball_tree, kulsinski | ball_tree, rogerstanimoto | ball_tree, russellrao | ball_tree, sokalmichener | ball_tree, sokalsneath | kd_tree, p | kd_tree, chebyshev | kd_tree, infinity | brute, cosine | brute, braycurtis | brute, canberra | brute, chebyshev | brute, correlation | brute, cosine | brute, dice | brute, hamming | brute, jaccard | brute, kulsinski | brute, mahalanobis | brute, matching | brute, rogerstanimoto | brute, russellrao | brute, seuclidean | brute, sokalmichener | brute, sokalsneath | brute, sqeuclidean | brute, yule

**Comment by prabhat00155** [comment]
https://github.com/onnx/sklearn-onnx/pull/240
